javascript/challenges/tree/tree.js

```diff
@@ -57,6 +57,22 @@ class BinaryTree {
     }
     return tally;
   }
+
+  breadthFirst(BinaryTree) {
+  let array = [];
+  let final
+  let current = this.root;
+  let level = 0;
+  let valueRead = false
+  if (!current) {
+    throw new error('the tree is empty');
+  } else {
+    while (current !== null)
+    array.push(current);
+    current.valueRead = true;
+    if (current.valueRead =)
+  }
+  
```

Comment:
You're off to a good start Garrett, when you have some time to come back to this I will regrade for you!